File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    for (let i = 0; i < nums.length; i++) {
        let a = nums[i];
        let b = target - a;
        let j = nums.indexOf(b);
        if (j !== -1 && j !== i) {
            return [i, j];
        }
    }
    return []; 
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mp;
        for (int i = 0; i < nums.size(); ++i) {
            int b = nums[i], a = target - b;
            if (mp.count(a))
                return {mp[a], i};
            mp[b] = i;
        }
        return {};
    }
};